dotCMS/src/main/java/com/dotcms/rest/api/v1/portlet/PortletResource.java

```diff
@@ -211,21 +213,62 @@ public final Response deletePortletForRole(@Context final HttpServletRequest req
 
     }
 
+    /**
+     * This endpoint returns a portlet's details given its id
+     * @param request
+     * @param portletId
+     * @return
+     */
+    @GET
+    @JSONP
+    @Path("/{portletId}")
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public final Response findPortlet(@Context final HttpServletRequest request,
+            @PathParam("portletId") final String portletId) {
+
+        try {
+            final InitDataObject initData = new WebResource.InitBuilder(webResource)
+                    .requiredBackendUser(true)
+                    .requiredFrontendUser(false)
+                    .requestAndResponse(request, null)
+                    .rejectWhenNoUser(true)
+                    .init();
+
+            final User user = initData.getUser();
+
+            if (!APILocator.getLayoutAPI().doesUserHaveAccessToPortlet(portletId, user)) {
+                return ResponseUtil.INSTANCE
+                        .getErrorResponse(request, Response.Status.UNAUTHORIZED, user.getLocale(),
+                                user.getUserId(), "unable to get portlet info");
+            }
+
+            return Response.ok(new ResponseEntityView(
+                    map("response", APILocator.getPortletAPI().findPortlet(portletId)))).build();
+        } catch (Exception e) {
+            return ResponseUtil.mapExceptionResponse(e);
+        }
+    }
+
     @GET
     @JSONP
     @Path("/{portletId}/_doesuserhaveaccess")
     @NoCache
     @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
     public final Response doesUserHaveAccessToPortlet(@Context final HttpServletRequest request,
-            @PathParam("portletId") final String portletId) throws DotDataException {
+            @PathParam("portletId") final String portletId) {
         final InitDataObject initData = new WebResource.InitBuilder(webResource)
                 .requiredBackendUser(true)
                 .requiredFrontendUser(false)
                 .requestAndResponse(request, null)
                 .rejectWhenNoUser(true)
                 .init();
-        return Response.ok(new ResponseEntityView(map("response", APILocator.getLayoutAPI()
-                .doesUserHaveAccessToPortlet(portletId, initData.getUser())))).build();
+        try {
+            return Response.ok(new ResponseEntityView(map("response", APILocator.getLayoutAPI()
```

Comment:
same than above

Reply:
Same comment as above. It isn't a new endpoint